haystack/core/component/component.py

```diff
@@ -184,6 +186,66 @@ def positional_to_kwargs(cls_type, args) -> Dict[str, Any]:
             out[name] = arg
         return out
 
+    @staticmethod
+    def _parse_and_set_output_sockets(instance: Any):
+        has_async_run = hasattr(instance, "async_run")
```

Comment:
I think it would be better if we go with `arun` instead of `async_run`, it's much more common and people are used to it.

Reply:
I considered it, but I prefer to keep names as explicit as possible. If people know what `arun` means, they'll certainly have no trouble understanding `async_run`.

Reply:
I wouldn't break the convention, I prefer we're predictable in this case.

Also `async_run` creates an unecessary duplication when declaring it though, `async def async_run`.

Reply:
After an internal discussion, we've decided to keep the naming for now and, if user feedback drives it, change when experimental async support gets merged into core.